src/components/Seo.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import { Helmet } from 'react-helmet'
+import { useLocation } from '@reach/router'
+import { useStaticQuery, graphql } from 'gatsby'
+
+interface Props {
+  title: string;
+  description: string;
+  image: string;
+  article: boolean;
+  author: string;
+  tags: string[];
+}
+
+const SEO: React.FC<Props> = (props: Props) => {
+  const {
+    title = null,
+    description = null,
+    image = null,
+    article = false,
+    author = null,
+    tags = null,
+  } = props
+  const { pathname } = useLocation()
+  const { site } = useStaticQuery(query)
+
+  const {
+    defaultTitle,
+    defaultDescription,
+    siteUrl,
+    defaultAuthor,
+  } = site.siteMetadata
+
+  const seo = {
+    title: title || defaultTitle,
+    description: description || defaultDescription,
+    image: `${siteUrl}${image}`,
+    url: `${siteUrl}${pathname}`,
+    author: author || defaultAuthor,
+    tags: tags && tags.join(','),
+  }
+
+  return (
+    <Helmet title={seo.title}>
+      <meta name="description" content={seo.description} />
+      <meta name="image" content={seo.image} />
+
+      {seo.url && <meta property="og:url" content={seo.url} />}
+
+      {(article ? true : null) && <meta property="og:type" content="article" />}
+      {(article ? true : null) && <meta property="og:article:tags" content={seo.tags} /> }
```

Comment:
```suggestion
      {(article ? true : null) && <meta property="og:article:tag" content={seo.tags} /> }
```

see https://ogp.me/#no_vertical